extensions/sovity-messenger/src/main/java/de/sovity/edc/extension/messenger/SovityMessengerRegistry.java

```diff
@@ -0,0 +1,102 @@
+     * @param handler         The function to process this type of message.
+     * @param <IN>            Incoming message type.
+     * @param <OUT>           Outgoing message type you send with {@link SovityMessenger#send(Class, String, SovityMessage)}.
+     */
+    @SneakyThrows
+    public <IN extends SovityMessage, OUT> void register(Class<IN> incomingMessage, Function<IN, OUT> handler) {
+        val type = getTypeViaIntrospection(incomingMessage);
+        register(incomingMessage, type, handler);
+    }
+
+    /**
+     * Registers a signal. This is a simplified version of a message where no answer is expected.
+     *
+     * @param incomingSignal The signal to send.
+     * @param handler        Signal processing
+     */
+    @SneakyThrows
+    public <IN extends SovityMessage> void registerSignal(Class<IN> incomingSignal, Consumer<IN> handler) {
+        val type = getTypeViaIntrospection(incomingSignal);
+        registerSignal(incomingSignal, type, handler);
+    }
+
+    /**
+     * Use this constructor only if your message can't have a default constructor. Otherwise, prefer using
+     * {@link #register(Class, Function)} for type safety.
+     */
+    public <IN extends SovityMessage, OUT> void register(Class<IN> clazz, String type, Function<IN, OUT> handler) {
+        if (handlers.containsKey(type)) {
+            throw new IllegalStateException("A handler is already registered for " + type);
+        }
+
+        handlers.put(type, new Handler<>(clazz, handler));
+    }
+
+    /**
+     * Use this constructor only if your message can't have a default constructor. Otherwise, prefer using
+     * {@link #registerSignal(Class, Consumer)} for type safety.
+     */
+    public <IN extends SovityMessage> void registerSignal(Class<IN> clazz, String type, Consumer<IN> handler) {
+        if (handlers.containsKey(type)) {
+            throw new IllegalStateException("A handler is already registered for " + type);
+        }
+        register(clazz, type, in -> {
+            handler.accept(in);
+            return null;
+        });
+    }
+
+    /**
+     * Retrieve a handler for the specified message type.
+     *
+     * @param type A unique ID to identify a message type.
+     * @return The function to process this message type.
+     */
+    @SuppressWarnings("unchecked")
+    public Handler<Object, Object> getHandler(String type) {
+        return (Handler<Object, Object>) handlers.get(type);
+    }
+
+    private static <IN extends SovityMessage> String getTypeViaIntrospection(Class<IN> incomingMessage) throws NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
+        val defaultConstructor = incomingMessage.getConstructor();
+        defaultConstructor.setAccessible(true);
+        val type = defaultConstructor.newInstance().getType();
+        return type;
+    }
```

Comment:
To do 
```java
handlers.register(Addition.class, in -> new Answer(in.getOp1() + in.getOp2()));
```
instead of 
```java
handlers.register(Addition.class, "add", in -> new Answer(in.getOp1() + in.getOp2()));
```

And avoid inconsistencies.